### DISCUSSION TITLE
about CAPE-OPEN interface standard

### DISCUSSION THREAD

**Discussion by Jackxi1024** [IDAES/idaes-pse#414] [Q&A] [answered]
CAPE-OPEN standard is an open interface standard in chemical process simulation software, which is supported by  Aspen Plus, HYSYS, Pro II software, etc.  
Does idaes support this interface standard ?

**Comment by andrewlee94** [comment, marked as answer]
No, IDAES does not support CAPE-OPEN and it is not currently on our development plan. Unfortunately (to our understanding at least), the CAPE-OPEN standard leaves a lot to interpretation and generally does not support the features we need for an fully Equation Oriented environment (notably full first and second partial derivatives).

**Reply by andrewlee94** [reply]
To expand a bit on my previous comment, what was your use case for a CAPE-OPEN interface? Did you want to include CAPE-OPEN model from another tool within IDAES, or did you want to include an IDAES model in another CAPE-OPEN tool?

As I mentioned before, the first is difficult to support and not on our current development plan. However, the second option might not be so hard to implement, but we have not looked into that either,

**Comment by Jackxi1024** [comment]
Thank you very much. I just saw the introduction of CAPE-OPEN interface in the literature and thought it was good, so I asked about the relationship between idaes and CAPE-OPEN. Now I know. Thank you very much




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "IDAES/idaes-pse"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月7日(星期三) 凌晨1:59
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [IDAES/idaes-pse] about CAPE-OPEN interface standard (#414)





 
To expand a bit on my previous comment, what was your use case for a CAPE-OPEN interface? Did you want to include CAPE-OPEN model from another tool within IDAES, or did you want to include an IDAES model in another CAPE-OPEN tool?
 
As I mentioned before, the first is difficult to support and not on our current development plan. However, the second option might not be so hard to implement, but we have not looked into that either,
